src/spritecache_internal.h

```diff
@@ -21,22 +21,31 @@
 /* These declarations are internal to spritecache but need to be exposed for unit-tests. */
 
 struct SpriteCache {
-	void *ptr;
-	size_t file_pos;
-	SpriteFile *file;    ///< The file the sprite in this entry can be found in.
-	uint32_t id;
-	int16_t lru;
-	SpriteType type;     ///< In some cases a single sprite is misused by two NewGRFs. Once as real sprite and once as recolour sprite. If the recolour sprite gets into the cache it might be drawn as real sprite which causes enormous trouble.
-	bool warned;         ///< True iff the user has been warned about incorrect use of this sprite
-	byte control_flags;  ///< Control flags, see SpriteCacheCtrlFlags
+	std::vector<byte> data{};
+	size_t file_pos{};
+	SpriteFile *file{};    ///< The file the sprite in this entry can be found in.
+	uint32_t id{};
+	uint32_t lru{};
+	SpriteType type{};     ///< In some cases a single sprite is misused by two NewGRFs. Once as real sprite and once as recolour sprite. If the recolour sprite gets into the cache it might be drawn as real sprite which causes enormous trouble.
+	bool warned{};         ///< True iff the user has been warned about incorrect use of this sprite
+	byte control_flags{};  ///< Control flags, see SpriteCacheCtrlFlags
```

Comment:
I know this is fine and all, but it still looks really odd to me to see it like this. I find the explicit style of saying `= false` and `= 0` much more verbose what the intention was. For vectors, sure, we don't have a choice. But PODs .. it is not my preference. But I am also not going to not approve it because of it :)